### ISSUE TITLE
Automatic reply  not work on DKIM signed domains

### ISSUE THREAD

**Issue by gizocz** [Mailu/Mailu#121]
I configured Auto-reply message in admin interface, but no message are replied.

Log:
imap_1       | Nov 22 10:51:32 lmtp(gizo@gitima.cz): Info: SuVyGjQjNFg6GAAABNQJ0Q: sieve: msgid=<CAGYk4S2TG-OLdKjAKO6ZgwzAHFjAW9WG9VK_6=WBYLa+0FvKZg@mail.gmail.com>: discarded duplicate vacation response to <gizocz@gmail.com>

version:
4067e3d95b43        mailu/dovecot:stable

**Comment by kaiyou** [comment]
This is actually the expected behaviour. The ``vacation`` Dovecot plugin only sends an automatic reply per day to each recipient to avoid spamming them with auto-replies.

The duration and count are customizable but customization from the admin interface is not implemented yet.

If you wish to reset the counter, simply delete the ``.dovecot.lda-dupes`` file in the user's maildir.

**Comment by gizocz** [comment]
The behaviour is OK, maybe a note in admin interface will be great.

But still not work:. Target smtp server reject the auto-reply message. It's
looks like an empty From header in auto-reply message:

smtp_1       | postfix/smtpd[1405]: 4801420969:
client=mailu_imap_1.mailu_default[172.19.0.3]
smtp_1       | postfix/cleanup[1403]: 4801420969: message-id=<
dovecot-sieve-1479814279-281626-0@mail.gitima.cz>
smtp_1       | postfix/qmgr[36]: 4801420969: from=<>, size=834, nrcpt=1
(queue active)
smtp_1       | postfix/smtp[1406]: 4801420969: host
postino.cesnet.cz[195.113.144.242]
said: 450 4.7.1 <mlazar@cesnet.cz>: Recipient address rejected: Greylisted
for 5 minutes (in reply to RCPT TO command)
smtp_1       | postfix/smtp[1406]: 4801420969: to=<mlazar@cesnet.cz>, relay=
cartero.cesnet.cz[195.113.144.196]:25, delay=6.7, delays=4.1/0/0.78/1.8,
dsn=2.0.0, status=sent (250 2.0.0 Ok: queued as EA2B66CA04F)
smtp_1       | postfix/qmgr[36]: 4801420969: removed

another smtp:
smtp_1       | postfix/smtpd[1405]: 77B6820969:
client=mailu_imap_1.mailu_default[172.19.0.3]
smtp_1       | postfix/cleanup[1403]: 77B6820969: message-id=<
dovecot-sieve-1479814210-458791-0@mail.gitima.cz>
smtp_1       | postfix/qmgr[36]: 77B6820969: from=<>, size=834, nrcpt=1
(queue active)
smtp_1       | postfix/smtp[1406]: 77B6820969: to=<gizocz@gmail.com>, relay=
gmail-smtp-in.l.google.com[64.233.166.26]:25, delay=4.5,
delays=4.1/0.02/0.24/0.21, dsn=5.7.1, status=bounced (host
gmail-smtp-in.l.google.com[64.233.166.26] said: 550-5.7.1 Unauthenticated
email from gitima.cz is not accepted due to domain's 550-5.7.1 DMARC
policy. Please contact the administrator of gitima.cz domain if 550-5.7.1
this was a legitimate mail. Please visit 550-5.7.1
https://support.google.com/mail/answer/2451690 to learn about the 550 5.7.1
DMARC initiative. d79si2062359wmi.16 - gsmtp (in reply to end of DATA
command))
smtp_1       | postfix/qmgr[36]: 77B6820969: removed

**Comment by kaiyou** [comment]
The issue here is related to DMARC:

```
gmail-smtp-in.l.google.com[64.233.166.26] said: 550-5.7.1 Unauthenticated
email from gitima.cz is not accepted due to domain's 550-5.7.1 DMARC
policy. Please contact the administrator of gitima.cz domain if 550-5.7.1
this was a legitimate mail. Please visit 550-5.7.1
https://support.google.com/mail/answer/2451690 to learn about the 550 5.7.1
DMARC initiative. d79si2062359wmi.16 - gsmtp (in reply to end of DATA
command))
```

Your DMARC entry is:

```
;; ANSWER SECTION:
_dmarc.gitima.cz.	600	IN	TXT	"v=DMARC1; p=reject; rua=mailto:dmarc-reports@gitima.cz; adkim=s; aspf=s"

```

Your SPF sounds fine:

```
gitima.cz.		600	IN	TXT	"v=spf1 mx a:mail.gitima.cz -all"
```

You also have a DKIM key configured:

```
;; ANSWER SECTION:
dkim._domainkey.gitima.cz. 600	IN	TXT	"v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDHucUduaDUBESozP7NBg3CuRYEUNxJfTaypBfcJiDvQDv1jbVQ4NSiVukP3JXgpgBBsLbydsOgLMQH3vUb5X2xMDXJJGCeR9Vwouabk3Xe7VviV0abnydwEu0VOYscW/nV6Uf4cYZeMR58oSt+WtaoENeKSp0bBd1Y7lclrskOewIDAQAB"
```

Are your emails accepted by Gmail if you send them using a mail client? If they are not, either there is general issue with DKIM signatures or you have a misconfiguration in your DNS entry or DKIM config.

If they are, maybe we have an issue with auto reply on DKIM-signed domains, I will have to test.

**Comment by gizocz** [comment]
Emails from webmail are accepted by gmail and SPF, DKIM, DMARC are marked
as PASS.

**Comment by kaiyou** [comment]
Ok, so emails sent from Pigeonhole do not used an authenticated SMTP session and are thus not signed by Rmilter. I will dig into the configuration a bit deeper.

**Comment by kaiyou** [comment]
I finally found the root cause: Pigeonhole was not setting the enveloppe address and rmilter was unable to guess which domain key the message should be signed with. Could you try with the latest Dovecot build once it is ready?

**Comment by gizocz** [comment]
Now it works. Tank you.

**Comment by gizocz** [comment]
It works for users emals. But not for an alias:

imap_1       | Nov 23 20:06:39 lmtp(gizo@gitima.cz): Info:
VrfOFs/2NVjpAAAAqusTng: sieve:
msgid=<CAGYk4S2_TiX9aWS2VmoqetMzjH5PkVDhgzadwUXXoeY6=efkgQ@mail.gmail.com>:
discarding vacation response for implicitly delivered message; no known
(envelope) recipient address found in message headers (recipient=<
gizo@gitima.cz>, and no additional `:addresses' are specified)

**Comment by kaiyou** [comment]
Thank you for finding this. It's a separate issue, also I just created #123, closing this one
